src/multi_auth/providers/google.cr

```diff
@@ -84,12 +84,18 @@ class MultiAuth::Provider::Google < MultiAuth::Provider
     @json = JSON.parse(raw_json)
     raise json["error"]["message"].as_s if json["error"]?
 
-    name = primary("names")
+    name = if primary?("names")
+      primary("names")
+    else
+      JSON::Any.new({} of String => JSON::Any)
```

Comment:
Using `JSON::Any` here instead of `nil` to keep the rest of the code the same. Otherwise the refactor would need to be a bit larger to avoid making `[]?` calls.